### PR TITLE
windows: cross-compile on linux

### PR DESCRIPTION
**Pull request by iameli**
Hopefully this will speed up the very slow Windows builds. I actually had this working at the same time as #2499 but didn't get a chance to clean it up until now.

@hjpotter92 Anything that needs to be done to accommodate the release/signing process? 

Tested on Windows 10 with both CPU and GPU transcoding:
image

### CODE REVIEW

**Review by thomshutt — APPROVED** [review]

File: docker/Dockerfile [hunk 1]
```diff
@@ -16,7 +18,7 @@ RUN	apt update \
 	&& curl -fsSl https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - \
 	&& add-apt-repository "deb [arch=amd64] https://apt.llvm.org/xenial/ llvm-toolchain-xenial-12 main" \
 	&& apt update \
-	&& apt -y install clang-12 clang-tools-12 lld-12 build-essential pkg-config autoconf git python docker-ce-cli pciutils gcc-multilib libgcc-8-dev-arm64-cross
+	&& apt -y install clang-12 clang-tools-12 lld-12 build-essential pkg-config autoconf git python docker-ce-cli pciutils gcc-multilib libgcc-8-dev-arm64-cross gcc-mingw-w64-x86-64
```
Comment (hjpotter92) [inline docker/Dockerfile:21]: would this help with docker images on windows based containers?

Comment (iameli) [reply docker/Dockerfile:21]: I tried messing around with Windows containers a while back and found them nearly impossible to use — incredibly slow and things froze up all the time. But maybe!

**Comment by iameli** [comment]
Before:
image

After:
image

🦾

**Comment by thomshutt** [comment]
> Before: image
> 
> After: image
> 
> 🦾

**Pull request merged**